jujugui/static/gui/src/app/store/env/api.js

```diff
@@ -2705,6 +2705,397 @@ YUI.add('juju-env-api', function(Y) {
+                  // Life cycle info.
+                  life: ""
+                }
+              }
+            },
+            // A list of relations established in this model.
+            relations: [
+              {
+                id: 1,
+                key: "mysql:cluster",
+                interface: "mysql-ha",
+                scope: "global",
+                endpoints: [{
+                  application: "mysql",
+                  name: "cluster",
+                  role: "peer",
+                  subordinate: false
+                }]
+              },
+              {
+                id: 3,
+                key: "wordpress:db mysql:db",
+                interface: "mysql",
+                scope: "global",
+                endpoints: [{
+                  application: "mysql",
+                  name: "db",
+                  role: "provider",
+                  subordinate: false
+                }, {
+                  application: "wordpress",
+                  name: "db",
+                  role: "requirer",
+                  subordinate: false
+                }]
+              },
+              {
+                id: 4,
+                key: "haproxy:reverseproxy wordpress:website",
+                interface: "http",
+                scope: "global",
+                endpoints: [{
+                  application: "wordpress",
+                  name: "website",
+                  role: "provider",
+                  subordinate: false
+                }, {
+                  application: "haproxy",
+                  name: "reverseproxy",
+                  role: "requirer",
+                  subordinate: false
+                }]
+              }
+            ]
+          }
+    */
+    fullStatus: function(callback) {
+      // Define the callback handler.
+      const handler = data => {
+        console.log(JSON.stringify(data));
+        if (!callback) {
+          console.log('data returned by Client.FullStatus API call:', data);
+          return;
+        }
+        if (data.error) {
+          callback(data.error, {});
+          return;
+        }
+        const resp = data.response;
+        if (typeof resp !== 'object' || !Object.keys(resp).length) {
+          callback('unexpected response: ' + JSON.stringify(resp), {});
+          return;
+        }
+        // Define some reusable helpers.
+        const URL = window.jujulib.URL;
+        const objMap = (obj, func) => {
+          if (!obj) {
+            return {};
+          }
+          return Object.keys(obj).reduce((collected, key) => {
+            const value = obj[key];
+            collected[key] = func(value);
+            return collected;
+          }, {});
+        };
+        const makeStatus = data => ({
+          status: data.status,
+          info: data.info,
+          life: data.life,
+          since: new Date(data.since)
+        });
+        const makeMachine = data => ({
+          id: data.id,
```

Comment:
Lots of repeated properties here, maybe we should investigate creating a `cloneButModifyTheseProperties` function 🤷‍♂️